src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/PreferStreamAsyncMemoryOverloads.cs

```diff
@@ -0,0 +1,304 @@
+                ParameterInfo.GetParameterInfo(cancellationTokenType)
+            };
+
+            // Retrieve the undesired methods
+            IMethodSymbol? undesiredReadAsyncMethod = readAsyncMethodGroup.GetFirstOrDefaultMemberWithParameterInfos(undesiredParameters);
+            if (undesiredReadAsyncMethod == null)
+            {
+                return;
+            }
+
+            IMethodSymbol? undesiredWriteAsyncMethod = writeAsyncMethodGroup.GetFirstOrDefaultMemberWithParameterInfos(undesiredParameters);
+            if (undesiredWriteAsyncMethod == null)
+            {
+                return;
+            }
+
+            IMethodSymbol? undesiredReadAsyncMethodWithCancellationToken = readAsyncMethodGroup.GetFirstOrDefaultMemberWithParameterInfos(undesiredParametersWithCancellationToken);
+            if (undesiredReadAsyncMethodWithCancellationToken == null)
+            {
+                return;
+            }
+
+            IMethodSymbol? undesiredWriteAsyncMethodWithCancellationToken = writeAsyncMethodGroup.GetFirstOrDefaultMemberWithParameterInfos(undesiredParametersWithCancellationToken);
+            if (undesiredWriteAsyncMethodWithCancellationToken == null)
+            {
+                return;
+            }
+
+            // Create the arrays with the exact parameter order of the undesired methods
+            var preferredReadAsyncParameters = new[]
+            {
+                ParameterInfo.GetParameterInfo(readOnlyMemoryType),  // ReadOnlyMemory<byte> buffer
+                ParameterInfo.GetParameterInfo(cancellationTokenType), // CancellationToken
+            };
+
+            var preferredWriteAsyncParameters = new[]
+            {
+                ParameterInfo.GetParameterInfo(memoryType),  // ReadOnlyMemory<byte> buffer
+                ParameterInfo.GetParameterInfo(cancellationTokenType), // CancellationToken
+            };
+
+            // Retrieve the preferred methods, which are used for constructing the rule message
+            IMethodSymbol? preferredReadAsyncMethod = readAsyncMethodGroup.FirstOrDefault(x =>
+                x.Parameters.Count() == 2 && x.Parameters[0].Type is INamedTypeSymbol type && type.ConstructedFrom.Equals(memoryType));
+            if (preferredReadAsyncMethod == null)
+            {
+                return;
+            }
+
+            IMethodSymbol? preferredWriteAsyncMethod = writeAsyncMethodGroup.FirstOrDefault(x =>
+                x.Parameters.Count() == 2 && x.Parameters[0].Type is INamedTypeSymbol type && type.ConstructedFrom.Equals(readOnlyMemoryType));
+            if (preferredWriteAsyncMethod == null)
+            {
+                return;
+            }
+
+            context.RegisterOperationAction(context =>
+            {
+                IInvocationOperation invocation = (IInvocationOperation)context.Operation;
+
+                IOperation parentOperation = invocation.Parent;
+                if (parentOperation == null)
```

Comment:
Nit: It would be much more readable to write this callback as a local function, say `ShouldAnalyze`. For example:

```csharp
context.RegisterOperationAction(context =>
{
   IInvocationOperation invocation = (IInvocationOperation)context.Operation;
   if (!ShouldAnalyze(invocation)
   {
   }

   ... remainder code ...
   DiagnosticDescriptor rule;
   string ruleMessageMethod;
   string ruleMessagePreferredMethod;

   ...
}

bool ShouldAnalyze(IInvocationOperation invocation)
{
   if (parentOperation == null)
   {
      return false;
   }

   if (parentOperation.Kind == OperationKind.Await)
   {
      return true;
   }

   ... other checks ...
}
}
```

Reply:
@mavasani Thanks. If this is a nit, then should it be a merge blocker, or can I get a sign off from you? I could address it in my next PR for the fixer.

Reply:
Note: The CI was saying there was a merge conflict with master so I just force pushed a rebase with the merge conflict fixed.